src/components/shared/AppBar/AppBar.js

```diff
@@ -0,0 +1,86 @@
+import React, {useEffect, useState} from 'react';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import InputBase from '@material-ui/core/InputBase';
+import SearchIcon from '@material-ui/icons/Search';
+import {Button, ButtonGroup} from "@material-ui/core";
+import {useDispatch} from "react-redux";
+import {loadBreeds} from "../../../store/actions/breedsReducerActions";
+import {useHistory, BrowserRouter as Router} from "react-router-dom";
+import {inputAction} from "../../../store/actions/inputAction";
+import {useStyles} from "./AppBarStyles";
+
+export function SearchAppBar() {
+    const classes = useStyles();
+    const dispatch = useDispatch();
+    useEffect(() => {
+        dispatch(loadBreeds());
+    }, [dispatch])
+
+    const [inputState, setInputState] = useState('');
+    const inputHandle = ({target}) => {
+        setInputState(target.value);
+    }
+    useEffect(() => {
+        dispatch(inputAction(inputState))
+    }, [dispatch, inputState])
+
+
+    let history = useHistory();
+
+    function galleryClick() {
+        history.push('/gallery')
+    }
+
+    function tableClick() {
+        history.push('/table')
+    }
+
+    function homeClick() {
+        history.push('/')
+    }
+
+    function favoritesClick() {
+        history.push('/favorites')
+    }
```

Comment:
Use arrow functions pls

Reply:
Moreover, use Link components as prop to button, and pass route using to="/somelink"